### ISSUE TITLE
[REQUEST/SUGGESTION] "Allow Up+Down or Left+Right" as retroarch option

### ISSUE THREAD

**Issue by retro-wertz** [libretro/RetroArch#6715]
# [REQUEST / SUGGESTION]:
the Up+Down or Left+Right button configuration (at least for dpad) should probably be one of the input options in retroarch, instead of a core option. My request comes from the idea that "normally", it should not be possible to press these opposing directions on the dpad on a normal controller.

im tagging the "input guy" for this :D @radius and probably others who can help

**Comment by i30817** [comment]
What's the point of this, to trigger errors in games? There were recently issues issued in the cores to prevent these very combinations (though i still think they should be prevented in the retroarch retropad side).

**Comment by retro-wertz** [comment]
i am not requesting to "allow" it (as worded).. (how stupid do you think I am) im requesting is as one of the retroarch option instead of having them on each and every core that behaves differently(for the better or worse) if they are allowed!

**Comment by i30817** [comment]
I don't think you're stupid man, the title triggered me because I was thinking of this and was a bit miffed the issue was moved to the cores.

So you'd like the same thing i'd want, a option in the input menu to allow 'pass up+down and left+right to the core' which defaults to 'no'? I agree it's much better than having each core 'deal with it', especially because they won't all 'deal with it' and the retropad is a gamepad abstraction and this never happens on a gamepad.

**Comment by i30817** [comment]
I'm thinking it's not a 'must have' but it might be useful in some games that have 'no latency' for stopping and take a while to accelerate. For example (not applicable to these controls i think) i was always annoyed in tomb raider when Lara stopped by accident and performed that long animation and i couldn't start running at the same speed immediately.

**Comment by i30817** [comment]
On second thought the savestate thing is a no-issue.

1. user savestates while having both directions pressed and '1 direction is winning'. Core saves the state as 'just the winning direction is pressed'.
2. user loads state as 'just the winning direction is pressed. As long as user has the right button pressed (ie the last direction), then there is no problem.

Because both buttons pressed is a momentary state, the user is not going to load a state while pressing both, so there is no inconsistency for the core. IF the user actually presses both while reloading a state, he's a idiot and deserves the stop/have a random direction as 'last pressed'.

**Comment by fr500** [comment]
@retro-wertz agreed, preventing this in the fronted would be far more optimal than doing it in every single core
